src/ClientTlsContext.php

```diff
@@ -366,6 +368,45 @@ public function getCertificate(): ?Certificate
         return $this->certificate;
     }
 
+    public function withPeerFingerprints(string|array $fingerprint): self
```

Comment:
Should we make this two separate methods, `withPeerFingerprint(string)` and `withPeerFingerprints(array)`?

Reply:
Sure, that seems a bit cleaner. Done!